addOns/authhelper/src/main/javahelp/org/zaproxy/addon/authhelper/resources/help/contents/browser-auth.html

```diff
@@ -8,23 +8,40 @@
 <BODY>
 <H1>Browser Based Authentication</H1>
 
-This <a href="authhelper.html">add-on</a> adds a new authentication type which uses a browser to login to the target website.
+This <a href="authhelper.html">add-on</a> adds a new authentication method which uses a browser to login to the target website.
 <p>
-It uses the first field with a type of "text" or "email" for the user name and the first field with a type
-of "password" for the password.
-It does support login pages where the password field is only made visible when the username has been entered.
-It does not support login pages which do not submit when <code>Return</code> is used in the password field.
+The authentication method requires the login URL and user credentials to perform the authentication. By default it automatically attempts to
+find suitable fields for the username and for the password, in the same page and multiple pages (i.e. password field is only made visible when
+the username has been entered).
 <p>
+The plan is to keep enhancing this authentication method to handle all of the most commonly used login page formats.
+If it does not work with your application's login page then please inform the ZAP team via the 
+<a href="https://groups.google.com/group/zaproxy-users">ZAP User Group</a> giving as many details as possible.
+
+<h3>Authentication Steps</h3>
+When the authentication method is not able to find suitable fields, or requires extra steps they can be manually specified by the user. The following steps are supported:
+<table>
+  <tr><th>Name</th><th>Type</th><th>Description</th></tr>
+  <tr><td>Auto Steps</td><td><code>AUTO_STEPS</code></td><td>Indicates that the authentication method should perform the steps that does by default, to identify and fill the required fields (if not already). This is useful when the authentication method is able to find and submit the fields but other required steps must happen before or after.</td></tr>
+  <tr><td>Click Element</td><td><code>CLICK</code></td><td>Clicks a web element, for example, to access the login form or dismiss a popup.</td>
+  <tr><td>Fill Custom Field</td><td><code>CUSTOM_FIELD</code></td><td>Fills a custom field with the given value, for example, choose a realm or organisation department in the login form.</td>
+  <tr><td>Fill Password Field</td><td><code>PASSWORD</code></td><td>Fills the password in a field, for when no suitable field was automatically identified.</td>
+  <tr><td>Fill TOTP Field</td><td><code>TOTP_FIELD</code></td><td>Fills a TOTP field.</td>
+  <tr><td>Fill Username Field</td><td><code>USERNAME</code></td><td>Fills the username in a field, for when no suitable field was automatically identified.</td>
```

Comment:
Not that I think we should change it but out of curiosity why not just allow the user to do multiple custom fields with whatever values instead of naming them specifically?

Reply:
To allow to use any user that might have been defined, if we hardcode credentials we can only use one.

Reply:
Okay